libqtile/layout/tile.py

```diff
@@ -105,43 +126,42 @@ def add(self, client, offset_to_current=0):
         elif self.add_on_top:
             self.clients.append_head(client)
         else:
-            self.clients.add(client, offset_to_current)
+            super().add(client, offset_to_current)
         self.reset_master()
 
     def configure(self, client, screen_rect):
+        if client not in self.clients:
+            return
         screen_width = screen_rect.width
         screen_height = screen_rect.height
         border_width = self.border_width
-        if self.clients and client in self.clients:
```

Comment:
As mentioned in the commit message, I don't see why this layout should hide foreign clients, while `Matrix` doesn't. If there is a good reason for hiding them, then I would have to make this change in `Matrix` instead.

Reply:
Maybe we should ask @cortesi, as it dates from [the original commit of this layout](https://github.com/qtile/qtile/commit/c45d2976a9742fb70ff2e1883e3a946a05fc78dc#diff-4adc59572ff217731264602873f7084fR114), 10 years ago. :)

I think it's safe to remove.

Reply:
Some of the other layouts check for foreign clients (and hide them), others don't. I guess there are multiple opinions on this.^^

Since the group is the one calling `Layout.layout()`, the group would have to be responsible for feeding the layouts with a wrong client. I don't think that can happen at the moment; and because of that, if it did, an exception might be preferrable over hiding a foreign client.
Maybe I should leave this section untouched?

Reply:
Yeah, let's keep it that way for now and make it consistent for all layouts in another PR if we think it's worth it.

Reply:
Ok, I removed that section.

Reply:
Don't forget to update the commit message. :)

Reply:
Oops. Also done.

Reply:
I just noticed that there is already a `@property` named `master_windows`. I'll have to come up with a different name then.

Reply:
Went for `master_length` since `master_size` seemed ambiguous and I found `master_stack_size` too long.